panic when using up arrow when filter has no matches

when using if you filter a choice such that there are no results and then press the up key then go panics.

using the default code from `Readme.md` in this repo

```
C:\workarea\source\github\jtnord\survey-panic >  type go.mod
module github.com/jtnord/survey-panic

require gopkg.in/AlecAivazis/survey.v1 v1.8.3

C:\workarea\source\github\jtnord\survey-panic >  type suvery.go
package main

import (
    "fmt"
    "gopkg.in/AlecAivazis/survey.v1"
)

// the questions to ask
var qs = []*survey.Question{
    {
        Name:     "name",
        Prompt:   &survey.Input{Message: "What is your name?"},
        Validate: survey.Required,
        Transform: survey.Title,
    },
    {
        Name: "color",
        Prompt: &survey.Select{
            Message: "Choose a color:",
            Options: []string{"red", "blue", "green"},
            Default: "red",
        },
    },
    {
        Name: "age",
        Prompt:   &survey.Input{Message: "How old are you?"},
    },
}

func main() {
    // the answers will be written to this struct
    answers := struct {
        Name          string                  // survey will match the question and field names
        FavoriteColor string `survey:"color"` // or you can tag fields to match a specific name
        Age           int                     // if the types don't match exactly, survey will try to convert for you
    }{}

    // perform the questions
    err := survey.Ask(qs, &answers)
    if err != nil {
        fmt.Println(err.Error())
        return
    }

    fmt.Printf("%s chose %s.", answers.Name, answers.FavoriteColor)
}
C:\workarea\source\github\jtnord\survey-panic >  go build

C:\workarea\source\github\jtnord\survey-panic >  survey-panic.exe
? What is your name? Bob
? Choose a color: silver  [Use arrows to move, space to select, type to filter]
panic: runtime error: index out of range

goroutine 1 [running]:
gopkg.in/AlecAivazis/survey%2ev1.(*Select).OnChange(0x6a0d60, 0x0, 0x0, 0x0, 0x0, 0x10, 0x6be700, 0x0, 0x0, 0x0, ...)
        C:/Users/jnord/go/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.3/select.go:152 +0x6c1
gopkg.in/AlecAivazis/survey%2ev1.(*Select).Prompt(0x6a0d60, 0x0, 0x0, 0x0, 0x0)
        C:/Users/jnord/go/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.3/select.go:232 +0x34d
gopkg.in/AlecAivazis/survey%2ev1.Ask(0x69c640, 0x3, 0x3, 0x546e60, 0xc00005cc00, 0x0, 0x0, 0x0, 0x580ef3, 0xc00006df88)
        C:/Users/jnord/go/pkg/mod/gopkg.in/!alec!aivazis/survey.v1@v1.8.3/survey.go:148 +0x191
main.main()
        C:/workarea/source/github/jtnord/survey-panic/suvery.go:39 +0x9a

C:\workarea\source\github\jtnord\survey-panic >  
```

update: occurs on windows and linux

--- a/select.go
+++ b/select.go
@@ -2,6 +2,7 @@ package survey
 
 import (
 	"errors"
+
 	"gopkg.in/AlecAivazis/survey.v1/core"
 	"gopkg.in/AlecAivazis/survey.v1/terminal"
 )
@@ -69,7 +70,7 @@ func (s *Select) OnChange(line []rune, pos int, key rune) (newLine []rune, newPo
 			return []rune(options[s.selectedIndex]), 0, true
 		}
 		// if the user pressed the up arrow or 'k' to emulate vim
-	} else if key == terminal.KeyArrowUp || (s.VimMode && key == 'k') {
+	} else if key == terminal.KeyArrowUp || (s.VimMode && key == 'k') && len(options) > 0 {
 		s.useDefault = false
 
 		// if we are at the top of the list
@@ -80,8 +81,9 @@ func (s *Select) OnChange(line []rune, pos int, key rune) (newLine []rune, newPo
 			// otherwise we are not at the top of the list so decrement the selected index
 			s.selectedIndex--
 		}
+
 		// if the user pressed down or 'j' to emulate vim
-	} else if key == terminal.KeyArrowDown || (s.VimMode && key == 'j') {
+	} else if key == terminal.KeyArrowDown || (s.VimMode && key == 'j') && len(options) > 0 {
 		s.useDefault = false
 		// if we are at the bottom of the list
 		if s.selectedIndex == len(options)-1 {

